src/figuresClasses.ts

```diff
@@ -1,11 +1,92 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green  ' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  readonly shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  readonly shape: Shape;
+
+  constructor(
+    public color: Color,
+    public a: number = 0,
+    public b: number = 0,
+    public c: number = 0,
+  ) {
+    this.shape = 'triangle';
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('some length is <= 0');
+    }
+
+    const sidesValue = [a, b, c];
+    const biggestNum = Math.max(...sidesValue);
+    const otherNumsSum = sidesValue
+      .filter((el) => el !== biggestNum)
+      .reduce((acc, current) => acc + current);
```

Comment:
Could be rewritten to `sidesValue.reduce((acc, current) => acc + current, -biggestNum)`